sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/Models/NetworkInterface.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+namespace Azure.ResourceManager.HardwareSecurityModules.Models
+{
+    /// <summary> The network interface definition. </summary>
+    public partial class NetworkInterface
+    {
+        /// <summary> Initializes a new instance of NetworkInterface. </summary>
+        public NetworkInterface()
+        {
+        }
+
+        /// <summary> Initializes a new instance of NetworkInterface. </summary>
+        /// <param name="id"> The ARM resource id in the form of /subscriptions/{SubscriptionId}/resourceGroups/{ResourceGroupName}/... </param>
+        /// <param name="privateIPAddress"> Private Ip address of the interface. </param>
+        internal NetworkInterface(string id, string privateIPAddress)
+        {
+            Id = id;
+            PrivateIPAddress = privateIPAddress;
+        }
+
+        /// <summary> The ARM resource id in the form of /subscriptions/{SubscriptionId}/resourceGroups/{ResourceGroupName}/... </summary>
+        public string Id { get; }
```

Comment:
What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#9f6fc0ae7c8e497e98d7979dadf97a4c).

Reply:
NetworkInterface Id is a string like "/subscriptions/{subscriptionId}/resourceGroups/{resourcegroupID}/providers/Microsoft.Network/networkInterfaces/{nicName}"

Reply:
> NetworkInterface Id is a string like "/subscriptions/{subscriptionId}/resourceGroups/{resourcegroupID}/providers/Microsoft.Network/networkInterfaces/{nicName}"

@emmeliaAra In that case, it should be a `ResourceIdentifier`. Need to change the format. See the guide link above.

Reply:
@archerzz I am not able to open the guide links but I was able to make the change based on examples from other projects. Please let me know if there is any issue with the changes.

Reply:
LGTM Thanks!